libs/sdk-ui-ext/src/internal/interfaces/VisualizationDescriptor.ts

```diff
@@ -0,0 +1,95 @@
+// (C) 2021 GoodData Corporation
+import { IInsightDefinition } from "@gooddata/sdk-model";
+import { IVisConstruct, IVisualization } from "./Visualization";
+
+/**
+ * Factories that create a new instance of pluggable visualization.
+ *
+ * @alpha
+ */
+export type PluggableVisualizationFactory = (params: IVisConstruct) => IVisualization;
+
+/**
+ * Info about min, max and default in layout units.
+ * If no value is provided for min/max/default layout will use its generic defaults for it.
+ *
+ * @alpha
+ */
+interface ISizeInfo {
+    default?: number;
+    min?: number;
+    max?: number;
+}
+
+/**
+ * Info about visualization's min, max and default for width and height.
+ *
+ * @alpha
+ */
+export interface IVisualizationSizeInfo {
+    width: ISizeInfo;
+    height: ISizeInfo;
+}
+
+/**
+ * Layout descriptor.
+ * Provides parameters of used layout.
+ * Width of the grid, height of one grid row and others
+ *
+ * @alpha
+ */
+export interface ILayoutDescriptor {
+    gridColumnsCount: number;
+    gridRowsCount: number;
+    gridColumnWidth: number;
+    gridRowHeight: number;
+    layoutContainerSize?: {
+        width: number;
+        height: number;
+    };
+    /**
+     * Converts width in px to the width in layout units
+     */
+    toGridWidth(widthPx: number): number;
+    /**
+     * Converts width in layout units to the width in px
+     */
+    toWidthInPx(width: number): number;
+    /**
+     * Converts height in px to the height in layout units
+     */
+    toGridHeight(heightPx: number): number;
+    /**
+     * Converts height in layout units to the height in px
+     */
+    toHeightInPx(height: number): number;
+}
+
+/**
+ * @alpha
+ * Visualization descriptor
+ * Provides access to the visualization's factory and additional details about visualization
+ */
+export interface IVisualizationDescriptor {
+    /**
+     * Gets pluggable visualization factory.
+     *
+     * @alpha
+     */
+    getFactory(): PluggableVisualizationFactory;
+
+    /**
+     * Gets info about size of visualization based on its type, bucket content, layout context and if custom height is enabled.
+     * Provided info is used by layout and it defines what will be the default size during render of insight and what will be the limits during resizing of insight.
+     *
+     * @param insight Current insight definition
+     * @param layoutDescriptor layout creating context for visualization
+     * @param enableCustomHeight if custom heights are allowed
```

Comment:
this enableCustomHeight is some kind of platform setting / feature flag?

if so, it may be better to pass the entire ISettings here